src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1703,6 +1720,39 @@ RouterJob(Query *originalQuery, PlannerRestrictionContext *plannerRestrictionCon
 		return job;
 	}
 
+	if (isMultiShardModifyQuery)
+	{
+		job->taskList = QueryPushdownSqlTaskList(originalQuery, job->jobId,
+												 plannerRestrictionContext->
+												 relationRestrictionContext,
+												 prunedShardIntervalListList,
+												 MODIFY_TASK,
+												 requiresMasterEvaluation);
+	}
+	else
+	{
+		GenerateSingleShardRouterTaskList(job, relationShardList,
+										  placementList, shardId);
+	}
+
+	job->requiresMasterEvaluation = requiresMasterEvaluation;
+	return job;
+}
+
+
+/*
+ * SingleShardRouterTaskList is a wrapper around other corresponding task
+ * list generation functions speficit to single shard selects and modifications.
```

Comment:
```suggestion
 * list generation functions speficic to single shard selects and modifications.
```

Reply:
fyi: I generally don't prefer committing on the remote branch via github, as it complicates rebases and could potentially got lost. I added this change locally.